pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -255,6 +259,94 @@ func (n *NetworkPolicyController) deleteNamespace(old interface{}) {
 	}
 }
 
+func (c *NetworkPolicyController) filterAGsFromNodeLabels(node *v1.Node) sets.String {
+	ags := sets.NewString()
+	addressGroupObjs, _ := c.addressGroupStore.GetByIndex(store.IsNodeAddressGroupIndex, "true")
+	for _, addressGroupObj := range addressGroupObjs {
+		addressGroup := addressGroupObj.(*antreatypes.AddressGroup)
+		nodeSelector := addressGroup.Selector.NodeSelector
+		if nodeSelector.Matches(labels.Set(node.GetLabels())) {
+			ags.Insert(addressGroup.Name)
+		}
+	}
+	return ags
+}
+
+func (c *NetworkPolicyController) addNode(obj interface{}) {
+	node := obj.(*v1.Node)
+	affectedAGs := c.filterAGsFromNodeLabels(node)
+	for key := range affectedAGs {
+		c.enqueueAddressGroup(key)
+	}
+	klog.V(2).InfoS("Processed Node CREATE event", "nodeName", node.Name, "affectedAGs", affectedAGs.Len())
+}
+
+func (c *NetworkPolicyController) deleteNode(obj interface{}) {
+	node, ok := obj.(*v1.Node)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Processed Node DELETE event error", "obj", obj)
+			return
+		}
+		node, ok = tombstone.Obj.(*v1.Node)
+		if !ok {
+			klog.ErrorS(nil, "Processed Node DELETE event error", "obj", tombstone.Obj)
+			return
+		}
+	}
+	// enqueue affected address group
+	affectedAGs := c.filterAGsFromNodeLabels(node)
+	for key := range affectedAGs {
+		c.enqueueAddressGroup(key)
+	}
+	klog.V(2).InfoS("Processed Node DELETE event", "nodeName", node.Name, "affectedAGs", affectedAGs.Len())
+}
+
+func nodeIPChanged(oldNode, newNode *v1.Node) (changed bool) {
+	var oldIPs, newIPs, oldNodeGWIPs, newNodeGWIPs *ip.DualStackIPs
+	var err error
+	if oldIPs, err = k8s.GetNodeAddrs(oldNode); err != nil {
+		return true
+	}
+	newIPs, err = k8s.GetNodeAddrs(newNode)
+	if err != nil || !reflect.DeepEqual(newIPs, oldIPs) {
+		return true
+	}
+	if oldNodeGWIPs, err = k8s.GetNodeGWIPs(oldNode); err != nil {
+		return true
+	}
+	if newNodeGWIPs, err = k8s.GetNodeGWIPs(newNode); err != nil {
+		return true
+	}
+	return !reflect.DeepEqual(oldNodeGWIPs, newNodeGWIPs)
```

Comment:
I think we should include transportIPAddress as well if it's different from nodeIPs.
And we should document the behavior in antrea-network-policy.md

Reply:
Updated.